File: breast_cancer_ct.cpp
#include <iostream>
#include <Eigen/Dense>
#include <vector>

using namespace Eigen;
using namespace std;

// Function to initialize a 10x10 matrix with 'O's
MatrixXd initializeMatrix(int size) {
    MatrixXd matrix = MatrixXd::Zero(size, size);
    for (int i = 0; i < size; ++i) {
        for (int j = 0; j < size; ++j) {
            matrix(i, j) = 'O';
        }
    }
    return matrix;
}

// Function to print the matrix
void printMatrix(const MatrixXd& matrix) {
    for (int i = 0; i < matrix.rows(); ++i) {
        for (int j = 0; j < matrix.cols(); ++j) {
            cout << (char)matrix(i, j) << " ";
        }
        cout << endl;
    }
}

// Function to simulate x-ray interaction with Janus spheres
void simulateXRay(MatrixXd& matrix, int startX, int startY, int endX, int endY) {
    int x = startX, y = startY;
    while (x != endX || y != endY) {
        if (matrix(x, y) == 'O') {
            matrix(x, y) = 'X';  // Mark x-ray path
        }
        x += (endX - startX) / abs(endX - startX);
        y += (endY - startY) / abs(endY - startY);
    }
}

int main() {
    int size = 10;
    MatrixXd matrix = initializeMatrix(size);
    printMatrix(matrix);

    // Simulate x-ray interaction
    simulateXRay(matrix, 0, 0, 9, 9);
    cout << "\nAfter X-ray simulation:\n";
    printMatrix(matrix);

    return 0;
}

File: breast_cancer_ct.rs
fn initialize_matrix(size: usize) -> Vec<Vec<char>> {
    let mut matrix = vec![vec!['O'; size]; size];
    matrix
}

fn print_matrix(matrix: &Vec<Vec<char>>) {
    for row in matrix {
        for cell in row {
            print!("{} ", cell);
        }
        println!();
    }
}

fn simulate_xray(matrix: &mut Vec<Vec<char>>, start_x: usize, start_y: usize, end_x: usize, end_y: usize) {
    let mut x = start_x;
    let mut y = start_y;
    let step_x = if end_x >= start_x { 1 } else { -1 } as isize;
    let step_y = if end_y >= start_y { 1 } else { -1 } as isize;
    
    while x != end_x || y != end_y {
        if matrix[x][y] == 'O' {
            matrix[x][y] = 'X';  // Mark x-ray path
        }
        if x != end_x {
            x = (x as isize + step_x) as usize;
        }
        if y != end_y {
            y = (y as isize + step_y) as usize;
        }
    }
}

fn main() {
    let size = 10;
    let mut matrix = initialize_matrix(size);
    print_matrix(&matrix);

    // Simulate x-ray interaction
    simulate_xray(&mut matrix, 0, 0, 9, 9);
    println!("\nAfter X-ray simulation:");
    print_matrix(&matrix);
}

